src/target/efm32.c

```diff
@@ -538,12 +544,12 @@ bool efm32_probe(target_s *t)
 {
 	/* Check if the OUI in the EUI is silabs or energymicro.
 	 * Use this to identify the Device Identification (DI) version */
-	uint8_t di_version = 1;
-	uint64_t oui24 = ((efm32_v1_read_eui64(t) >> 40U) & 0xffffffU);
-	if (oui24 == EFM32_V1_DI_EUI_SILABS) {
+	uint8_t di_version;
+	uint64_t oui24;
+	if (oui24 = ((efm32_v1_read_eui64(t) >> 40U) & 0xffffffU), oui24 == EFM32_V1_DI_EUI_SILABS) {
 		/* Device Identification (DI) version 1 */
 		di_version = 1;
-	} else if (oui24 == EFM32_V2_DI_EUI_ENERGYMICRO) {
+	} else if (oui24 = ((efm32_v2_read_eui48(t) >> 24U) & 0xffffffU), oui24 == EFM32_V2_DI_EUI_ENERGYMICRO) {
```

Comment:
Per the lint pass, do not do assignments in `if` statements - as this is necessary, split these out into two separate assignments and checks please.